token/link/link.go

```diff
@@ -256,9 +264,52 @@ func (service *LinkService) Callback(ctx context.Context, req *goa.RequestData,
 	return knownReferrer, nil
 }
 
-// NewOauthProvider creates a new oauth provider for the given resource URL
-func (service *OauthProviderFactoryService) NewOauthProvider(ctx context.Context, req *goa.RequestData, forResource string) (ProviderConfig, error) {
+// NewOauthProvider creates a new oauth provider for the given resource URL or provider alias
+func (service *OauthProviderFactoryService) NewOauthProvider(ctx context.Context, identityID uuid.UUID, req *goa.RequestData, forResource string) (ProviderConfig, error) {
 	authURL := rest.AbsoluteURL(req, "")
+
+	// Check if the forResource is actually a provider alias like "github" or "openshift"
+	if forResource == GitHubProviderAlias {
+		return NewGitHubIdentityProvider(service.config.GetGitHubClientID(), service.config.GetGitHubClientSecret(), service.config.GetGitHubClientDefaultScopes(), authURL), nil
+	}
+	if forResource == OpenShiftProviderAlias {
+		// Look up the user's OpenShift cluster
+		var clusterURL string
+		err := application.Transactional(service.db, func(appl application.Application) error {
+			identity, err := appl.Identities().Load(ctx, identityID)
```

Comment:
How about using the open ended query model `Query(IdentityWithUser(..))` ?

Reply:
Ah right. Done in https://github.com/fabric8-services/fabric8-auth/pull/335/commits/fbf9b742b9b1f8a8500cfdd8416e6fa7669d5641